src/containers/CheckoutPage/CheckoutPage.duck.js

```diff
@@ -177,9 +184,34 @@ export const initiateOrder = (orderParams, transactionId) => (dispatch, getState
     expand: true,
   };
 
-  const createOrder = transactionId ? sdk.transactions.transition : sdk.transactions.initiate;
+  if (!transactionId) {
+    return initiatePrivileged({ isSpeculative: false, bookingData, bodyParams, queryParams })
+      .then(response => {
+        const entities = denormalisedResponseEntities(response);
+        const order = entities[0];
+        dispatch(initiateOrderSuccess(order));
+        dispatch(fetchCurrentUserHasOrdersSuccess(true));
+        return order;
+      })
+      .catch(e => {
+        dispatch(initiateOrderError(storableError(e)));
+        const transactionIdMaybe = transactionId ? { transactionId: transactionId.uuid } : {};
+        log.error(e, 'initiate-order-failed', {
+          ...transactionIdMaybe,
+          listingId: orderParams.listingId.uuid,
+          bookingStart: orderParams.bookingStart,
+          bookingEnd: orderParams.bookingEnd,
+        });
+        throw e;
+      });
```

Comment:
This looks like the same code as repeated below. Extract to error handler function?